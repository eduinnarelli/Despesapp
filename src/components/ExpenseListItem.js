import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import locale from 'numeral/locales';

numeral.locale('pt-br');

const ExpenseListItem = ({id, description, amount, createdAt}) => (

  <div>

    <Link to={`/edit/${id}`}>
      <h4>{description}</h4>
    </Link>

    <p> 
      {numeral(amount / 100).format('$0,0.00')}
      - 
      {moment(createdAt).format('MMMM Do, YYYY')} 
    </p>

  </div>

);

export default ExpenseListItem;