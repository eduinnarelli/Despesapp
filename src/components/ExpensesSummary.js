import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import locale from 'numeral/locales';

numeral.locale('pt-br');

export const ExpensesSummary = (props) => {

  const expenseWord = props.expenses.length == 1 ? 'despesa' : 'despesas';
  const formattedExpensesTotal = numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00');

  return (

    <div>

      {

        <h4>
          Visualizando {props.expenses.length} {expenseWord},
          totalizando {' '} {formattedExpensesTotal}
        </h4>

      }

    </div>

  );
  
};

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpensesSummary); 
