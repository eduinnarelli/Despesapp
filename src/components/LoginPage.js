import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage = ({startLogin}) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Despes<span>{'{'}</span>app<span>{'}'}</span></h1>
      <button onClick={startLogin} className="button">Login com Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);