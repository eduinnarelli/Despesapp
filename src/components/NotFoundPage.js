import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
  <div className="box-layout box-layout--not-found">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Página não encontrada!</h1>
      <Link className="button" to="/">Voltar</Link>
    </div>
  </div>
);

export default NotFoundPage;