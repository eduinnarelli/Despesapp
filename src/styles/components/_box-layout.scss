.box-layout {
  align-items: center;
  background: url('/images/sad-doggo-2-adj.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout--not-found {
  background: url('/images/sad-doggo-adj.jpg');
  background-size: cover;
}

.box-layout__box {
  background: fade-out(white, .15);
  border-radius: 3px;
  padding: $l-size $m-size;
  text-align: center;
  width: 25rem;
}

.box-layout__title {

  margin: 0 0 $m-size 0;
  line-height: 1;
  padding-bottom: $m-size;

  span {
    color: $light-blue;
  }

}