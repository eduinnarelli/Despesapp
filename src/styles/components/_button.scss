.button {
  background: $blue;
  border: none;
  border-bottom: .3rem solid darken($blue, 10%);
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
  border: none;
  color: $off-white;
  font-size: 1.7rem;
  font-weight: 100;
  transition: color .3s ease;
  &:hover {
    color: $very-light-blue;
  }
}

.button--secondary {

  background: #888;
  border-bottom: .3rem solid darken(#888, 10%);
  margin-bottom: $m-size;

  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
  }

}