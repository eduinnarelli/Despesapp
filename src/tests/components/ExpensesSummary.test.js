import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render empty ExpensesSummary without expenses', () => {

  const wrapper = shallow(<ExpensesSummary expenses={[]}/>);
  expect(wrapper).toMatchSnapshot();

});

test('should render ExpensesSummary message with one expense', () => {

  const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]}/>);
  expect(wrapper).toMatchSnapshot();

});

test('should render ExpensesSummary message with multiple expenses', () => {

  const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();

});

