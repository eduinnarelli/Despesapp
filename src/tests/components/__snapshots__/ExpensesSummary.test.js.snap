// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary message with multiple expenses 1`] = `
<div>
  <h4>
    Visualizando 
    3
     
    despesas
    , totalizando 
     
     
    R$1.141,95
  </h4>
</div>
`;

exports[`should render ExpensesSummary message with one expense 1`] = `
<div>
  <h4>
    Visualizando 
    1
     
    despesa
    , totalizando 
     
     
    R$1.095,00
  </h4>
</div>
`;

exports[`should render empty ExpensesSummary without expenses 1`] = `
<div>
  <h4>
    Visualizando 
    0
     
    despesas
    , totalizando 
     
     
    R$0,00
  </h4>
</div>
`;
